jni/src/faiss_wrapper.cpp

```diff
@@ -28,9 +28,33 @@
 
 // Defines type of IDSelector
 enum FilterIdsSelectorType{
-    BITMAP, BATCH
+    BITMAP = 0, BATCH = 1,
 };
-
+namespace faiss {
+struct IDSelectorJlongBitmap : IDSelector {
+    size_t n;
+    const jlong* bitmap;
+
+    /** Construct with a binary mask like Lucene FixedBitSet
+     *
+     * @param n size of the bitmap array
+     * @param bitmap id like Lucene FixedBitSet bits
+     */
+    IDSelectorJlongBitmap(size_t n, const jlong* bitmap) : n(n), bitmap(bitmap) {};
+    bool is_member(idx_t id) const final {
+        uint64_t index = id;
+        uint64_t i = index >> 6;  // div 64
+        if (i >= n ) {
+            return false;
+        }
+        // signed shift will keep a negative index and force an
+        // array-index-out-of-bounds-exception, removing the need for an explicit check.
+        jlong bitmask = 1L << index;
```

Comment:
This doesn't seem right. If this code is copied from Lucene which is java implementation then this will not work.

Do this:
`(1 << (index & 63))` to make sure that mask is correct. I have seen this earlier also in @heemin32 code. The code you added works well in java but not in c++. Please add UTs with values like 84, 65 and see this mask you are creating is correct. Last I checked it doesn't work.

Reply:
LGTM